xivo/wsgi.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2014-2024 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import logging
+import time
+
+from cheroot.wsgi import PathInfoDispatcher, Server
+
+logger = logging.getLogger(__name__)
+
+"""
+    Patch for WAZO-3846 (https://wazo-dev.atlassian.net/browse/WAZO-3846)
+
+    After an OSError, cheroot worker threads would hang and any following
+    request opens a file descriptor until the maximum limit is reached. At this
+    point the logger outputs 5k lines/second of warnings about reaching the
+    limit of file descritors.
+
+    Stopping the service when such error occurs avoid flooding the log files.
+
+    Check upstream package for any change in the 'serve' function when migrating
+    from bullseye to bookworm (and subsequent)
+"""
```

Comment:
Maybe a nitpick, but I think comment syntax(`# ...`) is more relevant here.

top-level strings are used as "docstrings" only when they are the first value of a module, class or function. They are explicitly interpreted by the python interpreter and stored in object attributes(`__doc__`).

A module/class/function docstring is appropriate for information that can be discovered by documentation-generation tooling(which we don't use right now), but in this case I think a comment is more appropriate.

Reply:
Updated